avail-intellij-plugin/src/main/kotlin/org/availlang/intellij/plugin/module/AvailProjectTemplate.kt

```diff
@@ -0,0 +1,244 @@
+      config.selectedLibrary?.let { lib ->
+        configureAvailStdLib(config, lib)
+      }
+
+      updateBackingProjectFile(projectFilePath, config)
+    }
+  }
+
+  /**
+   * Updates the backing project file with the current AvailProjectV1 instance.
+   *
+   * @param projectFilePath The file path of the backing project file.
+   * @param config The configuration settings.
+   */
+  private fun AvailProjectV1.updateBackingProjectFile(
+    projectFilePath: String,
+    config: Config
+  )
+  {
+    if (projectFilePath.isNotEmpty()) {
+      // Update the backing project file.
+      val writer = JSONWriter.newPrettyPrinterWriter()
+      writeTo(writer)
+      File(projectFilePath).writeText(writer.contents())
+      config.environmentSettings.add(this, projectFilePath)
+    }
+  }
+
+  /**
+   * Configures the Avail standard library for the given Avail project.
+   *
+   * @param config The configuration settings for the Avail project.
+   * @param lib The file representing the Avail standard library.
+   */
+  private fun AvailProjectV1.configureAvailStdLib(
+    config: Config,
+    lib: File
+  )
+  {
+    val libName = config.libraryName ?: AVAIL_STDLIB_ROOT_NAME
+    val libConfigDir = AvailEnvironment.projectRootConfigPath(
+      config.fileName, libName, config.projectLocation
+    )
+    optionallyInitializeConfigDirectory(libConfigDir)
+    val jar = AvailArtifactJar(lib.toURI())
+
+    val rootManifest = jar.manifest.roots[AVAIL_STDLIB_ROOT_NAME]
+    var sg = importStyles(config, jar)
+    var tg = importTemplates(config, jar)
+    val extensions = mutableListOf<String>()
+    var description = ""
+    rootManifest?.let {
+      sg = it.styles
+      tg = it.templates
+      extensions.addAll(it.availModuleExtensions)
+      description = it.description
+    }
+    val stdLib = AvailProjectRoot(
+      libConfigDir,
+      config.projectLocation,
+      libName,
+      AvailLibraries(
+        "org/availlang/${lib.name}",
+        Scheme.JAR,
+        AVAIL_STDLIB_ROOT_NAME
+      ),
+      LocalSettings(
+        AvailEnvironment.projectRootConfigPath(
+          config.fileName,
+          libName,
+          config.projectLocation
+        )
+      ),
+      sg,
+      tg,
+      extensions
+    )
+    stdLib.description = description
+    stdLib.saveLocalSettingsToDisk()
+    stdLib.saveTemplatesToDisk()
+    stdLib.saveStylesToDisk()
+    addRoot(stdLib)
+  }
+
+  /**
+   * Imports the templates from the given AvailArtifactJar and returns them as a TemplateGroup.
+   *
+   * @param config The configuration object that specifies whether to import templates.
+   * @param jar The AvailArtifactJar from which to import the templates.
+   * @return The imported templates as a TemplateGroup.
+   */
+  private fun importTemplates(
+    config: Config,
+    jar: AvailArtifactJar
+  ) = if (config.importTemplates)
+  {
+    jar.manifest.templatesFor(AVAIL_STDLIB_ROOT_NAME)
+      ?: TemplateGroup()
```

Comment:
I don't think Avail has a preference one way or another, but this could fit on one line (_less than 80 characters_):
```kotlin
jar.manifest.templatesFor(AVAIL_STDLIB_ROOT_NAME) ?: TemplateGroup()
```